src/app/detail/[task_id]/[user_type]/page.tsx

```diff
@@ -76,6 +76,8 @@ export default async function TaskDetailPage({
     throw new Error('Please provide a Valid Token')
   }
 
+  console.info(`app/detail/${task_id}/${user_type}/page.tsx | Serving user ${token} with payload`, tokenPayload)
+
```

Comment:
More than doing this in multiple places in the server components (page.tsx), why don't we create another server component that handles logging for all the server components and use that in the layout.tsx? 

For the CopilotAPI, it looks good.

Reply:
Great point... but as you can see we're logging the token payload as well so its easier to debug auth issues later on. And token payload is currently being fetched in the page level only. That's why I'm replicating those `console.info`s on all pages.

Same for not using `middleware.ts`